packages/expo-image/src/ExpoImage.web.tsx

```diff
@@ -57,50 +62,149 @@ function getObjectPositionFromContentPosition(contentPosition?: ImageContentPosi
+        ...commonStyles,
+      },
+      placeholder: {
+        opacity: state === 'loaded' ? '0' : '1',
+        ...commonStyles,
+      },
+    };
+  } else if (effect === ImageTransitionEffect.FLIP_FROM_TOP) {
+    const commonStyles = {
+      transition: `transform ${duration}ms`,
+      transformOrigin: 'top',
+      transitionTimingFunction: getCSSTiming(timing),
+    };
+    return {
+      placeholder: {
+        transform: `rotateX(${state !== 'loaded' ? '0' : '90deg'})`,
+        ...commonStyles,
+      },
+      image: {
+        transform: `rotateX(${state === 'loaded' ? '0' : '90deg'})`,
+        ...commonStyles,
+      },
+    };
+  }
+
+  return { placeholder: {}, image: {} };
+}
+
 export default function ExpoImage({
   source,
-  defaultSource,
+  placeholder,
   loadingIndicatorSource,
   contentPosition,
   onLoad,
+  transition,
   onLoadStart,
   onLoadEnd,
   onError,
   ...props
 }: ImageProps) {
   const { aspectRatio, backgroundColor, transform, borderColor, ...style } = props.style ?? {};
+  const [state, handlers] = useImageState(source);
+  const { placeholder: placeholderStyle, image: imageStyle } = useTransition(transition, state);
+
   const resolvedSources = ensureIsArray(source).map(resolveAssetSource);
+
   return (
-    <>
-      <picture
+    <div
+      style={{
+        aspectRatio: String(aspectRatio),
+        backgroundColor: backgroundColor?.toString(),
+        transform: transform?.toString(),
+        borderColor: borderColor?.toString(),
+        ...style,
+        overflow: 'hidden',
+        position: 'relative',
+      }}>
+      <img
+        src={ensureIsArray(placeholder).map(resolveAssetSource)?.[0]?.uri}
+        style={{
+          width: '100%',
+          height: '100%',
+          position: 'absolute',
+          left: 0,
+          right: 0,
+          objectFit: 'scale-down',
+          objectPosition: 'center',
+          ...placeholderStyle,
+        }}
+      />
+      <img
+        src={resolvedSources.at(0)?.uri}
         style={{
-          overflow: 'hidden',
-          ...style,
-        }}>
-        <img
-          src={resolvedSources.at(0)?.uri}
-          style={{
-            width: '100%',
-            height: '100%',
-            aspectRatio: String(aspectRatio),
-            backgroundColor: backgroundColor?.toString(),
-            transform: transform?.toString(),
-            borderColor: borderColor?.toString(),
-            objectFit: resolveContentFit(props.contentFit, props.resizeMode),
-            objectPosition: getObjectPositionFromContentPosition(contentPosition) || undefined,
-          }}
-        />
-      </picture>
-    </>
+          width: '100%',
+          height: '100%',
+          position: 'absolute',
+          left: 0,
+          right: 0,
+          objectFit: resolveContentFit(props.contentFit, props.resizeMode),
+          objectPosition: getObjectPositionFromContentPosition(contentPosition) || undefined,
```

Comment:
Can we please not use `undefined` here? 😅 This word should be prohibited

Reply:
So maybe `|| '50% 50%'`? This is the default value. Also `initial` or `inherit` should be fine I guess 🤔

Reply:
hmm, my thinking is that all these values place burden on the developer to make sure the value is the default one, and undefined is equal to the behaviour of the framework if the style was never set – it places the blame on the framework.

Don't feel too strongly about it though, so happy to do this the way you prefer.